package/MDAnalysis/transformations/boxdimensions.py

```diff
@@ -80,15 +99,20 @@ def __init__(self,
                        'np.float32 numpy.ndarray')
             raise ValueError(errmsg)
         try:
-            self.dimensions = self.dimensions.reshape(6, )
+            self.dimensions = self.dimensions.reshape(-1, 6)
         except ValueError:
             errmsg = (f'{self.dimensions} array does not have valid box '
                        'dimension shape.\nSimulation box dimensions are '
-                       'given by an float array of shape (6, ), '
-                       ' containing 3 lengths and 3 angles: '
+                       'given by an float array of shape (6, 0), (1, 6), '
+                       'or (N, 6) where N is the number of frames in the '
+                       'trajectory and the dimension vector(s) containing '
+                       '3 lengths and 3 angles: '
                        '[a, b, c, alpha, beta, gamma]')
             raise ValueError(errmsg)
 
     def _transform(self, ts):
-        ts.dimensions = self.dimensions
+        try:
+            ts.dimensions = self.dimensions[0] if self.dimensions.shape[0] == 1 else self.dimensions[ts.frame]
```

Comment:
This is both a long line and also a bit hard to read (at least to me), splitting this over a few lines probably would be worth it?